R/add_overall_ard.R

```diff
@@ -0,0 +1,87 @@
+#' ARD add overall column
+#'
+#' Adds a column with overall summary statistics to tables
+#' created by `tbl_ard_summary()`.
+#'
+#' @param x (`tbl_ard_summary`)\cr
+#'   A stratified 'gtsummary' table
+#' @param cards (`card`)\cr
+#'   An ARD object of class `"card"` typically created with `cards::ard_*()` functions.
+#' @param last (scalar `logical`)\cr
+#'   Logical indicator to display overall column last in table.
+#'   Default is `FALSE`, which will display overall column first.
+#' @param col_label (`string`)\cr
+#'   String indicating the column label. Default is `"**Overall**"`
+#' @param statistic ([`formula-list-selector`][syntax])\cr
+#'   Override the statistic argument in initial `tbl_*` function
+#'   call. Default is `NULL`.
+#' @param digits ([`formula-list-selector`][syntax])\cr
+#'   Override the digits argument in initial `tbl_*` function
+#'   call. Default is `NULL`.
+#' @inheritParams rlang::args_dots_empty
+#'
+#' @author Daniel D. Sjoberg
+#' @name add_overall_ard
+#' @return A `gtsummary` of same class as `x`
+#' @examples
+#' # Example 1 ----------------------------------
+#' # build primary table
+#' tbl <-
+#'   cards::ard_stack(
+#'     trial,
+#'     .by = trt,
+#'     cards::ard_continuous(variables = age),
+#'     cards::ard_categorical(variables = grade),
+#'     .missing = TRUE,
+#'     .attributes = TRUE,
+#'     .total_n = TRUE
+#'   ) |>
+#'   tbl_ard_summary(by = trt)
+#'
+#' # create ARD with overall results
+#' ard_overall <-
+#'   cards::ard_stack(
+#'     trial,
+#'     cards::ard_continuous(variables = age),
+#'     cards::ard_categorical(variables = grade),
+#'     .missing = TRUE,
+#'     .attributes = TRUE,
+#'     .total_n = TRUE
+#'   )
+#'
+#' # add an overall column
+#' tbl |>
+#'   add_overall(cards = ard_overall)
```

Comment:
It seems that this function is more adding one column than overall, because you need to specify that column with ard_stack definition. Would it be better to have one add_column function with this structure and one add_overall that takes the relevant info from input ard? like instead of doing tbl |> add_overall more doing ard_stack |> add_overall |> tbl_ard_summary.
Maybe both options?

Reply:
A general function for adding statistics from ARD is on the roadmap. (#1614 )

Also, for easier use, I think we'll also make this easier directly from `tbl_ard_summary()` (#1940)